### PR TITLE
Fixups for search results

### PR DESCRIPTION
**Pull request by cmc333333**
* Removed markerless paragraphs from the displayed label
* Removes subparts/subject groups from the search results (as we don't display whole subparts/groups in the UI)

Old:
screen shot 2015-09-10 at 1 26 29 pm

New:
screen shot 2015-09-10 at 1 27 58 pm

### CODE REVIEW

File: regulations/generator/node_types.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
-#vim: set encoding=utf-8
+# vim: set encoding=utf-8
 from itertools import takewhile
+import re
 
 APPENDIX = u'appendix'
 INTERP = u'interp'
 REGTEXT = u'regtext'
 SUBPART = u'subpart'
+SUBJGRP = u'subjgrp'
```
Comment (khandelwal) [inline regulations/generator/node_types.py]: A comment indicating what this is (in terms of a regulation would be useful here). That's probably true of the other things in this list too, but let's start here.

File: regulations/generator/node_types.py [hunk 2]
```diff
@@ -45,6 +49,22 @@ def label_to_text(label, include_section=True, include_marker=False):
...
+
+def _paragraph_tail(label_parts, join_with, prefix='', suffix=''):
+    """Given the tail of paragraph markers in a label, convert them into a
+    string, separated by the appropriate strings (join_with). Also remove any
+    markers following a markerless paragraph"""
+    not_markerless = lambda l: not _MARKERLESS_REGEX.match(l)
+    label_parts = list(takewhile(not_markerless, label_parts))
+    if label_parts:
+        return prefix + join_with.join(label_parts) + suffix
+        pass
```
Comment (khandelwal) [inline regulations/generator/node_types.py]: I don't think this pass will ever execute right?

File: regulations/generator/node_types.py [hunk 3]
```diff
@@ -45,6 +49,22 @@ def label_to_text(label, include_section=True, include_marker=False):
             or _l2t_section(label, include_section, include_marker))
 
 
+_MARKERLESS_REGEX = re.compile(r'p\d+')
+
+
+def _paragraph_tail(label_parts, join_with, prefix='', suffix=''):
+    """Given the tail of paragraph markers in a label, convert them into a
```
Comment (khandelwal) [inline regulations/generator/node_types.py]: Maybe this function should be called something like _join_paragraph_tail()

File: regulations/generator/node_types.py [hunk 4]
```diff
@@ -45,6 +49,22 @@ def label_to_text(label, include_section=True, include_marker=False):
             or _l2t_section(label, include_section, include_marker))
 
 
+_MARKERLESS_REGEX = re.compile(r'p\d+')
+
+
+def _paragraph_tail(label_parts, join_with, prefix='', suffix=''):
```
Comment (khandelwal) [inline regulations/generator/node_types.py]: This doesn't have any tests associated directly with it, which means it's hard to know what it's supposed to do.

File: regulations/generator/node_types.py (hunk 2 above)
Comment (cmc333333) [inline regulations/generator/node_types.py]: You're right, I'll remove

File: regulations/generator/node_types.py (hunk 3 above)
Comment (cmc333333) [inline regulations/generator/node_types.py]: Can do

File: regulations/generator/node_types.py (hunk 1 above)
Comment (cmc333333) [inline regulations/generator/node_types.py]: Sure thing

File: regulations/generator/node_types.py (hunk 4 above)
Comment (cmc333333) [inline regulations/generator/node_types.py]: I think that tests should focus on public functions; this private function is an implementation detail. I don't think it's necessary to establish via a test unless there's a known corner case.

Regarding what it's supposed to do, that's why there's a comment :)

**Comment by cmc333333** [comment]
Updated to address @khandelwal's comments
